api-admin/src/main/java/bio/terra/pearl/api/admin/service/study/StudyExtService.java

```diff
@@ -62,6 +69,29 @@ public Study create(String portalShortcode, StudyCreationDto study, AdminUser op
     return newStudy;
   }
 
+  @Transactional
+  public void delete(String portalShortcode, String studyShortcode, AdminUser operator) {
+    if (!operator.isSuperuser()) {
+      throw new PermissionDeniedException("You do not have permission to create studies");
+    }
+    NotFoundException notFoundException =
+        new NotFoundException(
+            "Portal "
+                + portalShortcode
+                + " was not found as one of the portals for study "
+                + studyShortcode);
+    Portal portal =
+        portalService
+            .findOneByShortcodeOrHostname(portalShortcode)
+            .orElseThrow(() -> notFoundException);
```

Comment:
I don't think you need to reuse the same exception across both cases
```suggestion
            .orElseThrow(() -> new NotFoundException(
            "Portal % was not found.".formatted(portalShortcode, studyShortcode);
```